### DISCUSSION TITLE
Iterate over command line arguments in bpftrace

### DISCUSSION THREAD

**Discussion by Rvdixit23** [bpftrace/bpftrace#2113] [Q&A] [answered]
I want to iterate over command line arguments
The [Reference Guide on Positional Arguments](https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md#9-1--n--positional-parameters) doesn't specify how to do this.
Here's what I tried
`$i=1;
while($i<=$#)
{printf("%d\n",$($i));
$i = $i + 1 ;}`
This doesn't seem to be working. I've tried changing doing it with and without brackets. But it says 'invalid character found'. I also looked through to see if any builtin function could help me but to no avail. Could someone please tell me how to do this?

**Comment by fbs** [comment, marked as answer]
that is currently not possible. Command line arguments are substituted at compile time and a while loop is a bpf runtime thing. 

It would be nice if we could find a way to iterate over maps, once we have that we might be able to add CLI arguments in a map
